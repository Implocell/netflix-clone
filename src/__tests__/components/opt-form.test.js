import { render } from "@testing-library/react";
import { OptForm } from "../../components";

describe("<OptForm />", () => {
	test("should render the <OptForm /> with populated data", () => {
		const { container, getByText, getByPlaceholderText } = render(
			<OptForm>
				<OptForm.Input placeholder="Email adress" />
				<OptForm.Button>Try it now</OptForm.Button>
				<OptForm.Text>
					Ready to watch? Enter your email to create or restart your membership
				</OptForm.Text>
			</OptForm>
		);

		expect(getByText("Try it now")).toBeTruthy();
		expect(
			getByText(
				"Ready to watch? Enter your email to create or restart your membership"
			)
		).toBeTruthy();
		expect(getByPlaceholderText("Email adress")).toBeTruthy();
		expect(container.firstChild).toMatchSnapshot();
	});
});
