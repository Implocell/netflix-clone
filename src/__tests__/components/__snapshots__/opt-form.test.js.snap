// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OptForm /> should render the <OptForm /> with populated data 1`] = `
<div
  class="opt-form__Container-sc-8oudnm-0 fZsAhm"
>
  <input
    class="opt-form__Input-sc-8oudnm-1 flzjTn"
    placeholder="Email adress"
  />
  <button
    class="opt-form__Button-sc-8oudnm-2 kIhYnw"
  >
    Try it now
     
    <img
      alt="try now"
      src="/images/icons/chevron-right.png"
    />
  </button>
  <p
    class="opt-form__Text-sc-8oudnm-3 cgZRtx"
  >
    Ready to watch? Enter your email to create or restart your membership
  </p>
</div>
`;
